lib/alloc.js

```diff
@@ -0,0 +1,40 @@
+class CodeAllocator {
+  constructor (sliceSize) {
+    this.sliceSize = sliceSize;
+    this.slicesPerPage = Process.pageSize / sliceSize;
+
+    this.pages = [];
+    this.free = [];
+  }
+
+  allocateSlice () {
+    let slice = this.free.pop();
```

Comment:
```suggestion
    const slice = this.free.pop();
```

As it's not re-assigned.